src/sonymn_int.cpp

```diff
@@ -635,109 +635,75 @@ constexpr TagDetails sonyLongExposureNoiseReduction[] = {
   auto pos = metadata->findKey(ExifKey("Exif.Image.Model"));
-  if (pos != metadata->end() && pos->size() != 0 && pos->typeId() == asciiString) {
-    val = pos->toString(0);
-    return true;
-  }
+  if (pos != metadata->end() && pos->size() != 0 && pos->typeId() == asciiString)
+    return pos->toString(0);
 
   // NOTE: As using the translated SonyModelID value, need to be synchronized with the array format
   pos = metadata->findKey(ExifKey("Exif.Sony1.SonyModelID"));
   if (pos != metadata->end() && pos->size() != 0 && pos->typeId() == unsignedShort) {
-    if (auto temp = pos->print(metadata); !Internal::contains(temp, ' ')) {
-      val = temp;
-      return true;
-    }
-    val = "";
-    return false;
+    if (auto temp = pos->print(metadata); !Internal::contains(temp, ' '))
+      return temp;
+    return "";
   }
   pos = metadata->findKey(ExifKey("Exif.Sony2.SonyModelID"));
   if (pos != metadata->end() && pos->size() != 0 && pos->typeId() == unsignedShort) {
-    if (auto temp = pos->print(metadata); !Internal::contains(temp, ' ')) {
-      val = temp;
-      return true;
-    }
-    val = "";
-    return false;
+    if (auto temp = pos->print(metadata); !Internal::contains(temp, ' '))
+      return temp;
+    return "";
   }
-
-  val = "";
-  return false;
+  return "";
 }
 
-static auto getAFAreaModeSetting(const ExifData* metadata, uint32_t& val) {
+static uint32_t getAFAreaModeSetting(const ExifData* metadata) {
   auto pos = metadata->findKey(ExifKey("Exif.Sony1.AFAreaModeSetting"));
   if (pos != metadata->end() && pos->size() != 0 && pos->typeId() == unsignedByte) {
     std::ostringstream oss;
     pos->write(oss, metadata);
-    if (oss.str() == _("n/a")) {
-      val = 0;
-      return false;
-    }
-
-    val = pos->toUint32(0);
-    return true;
+    if (oss.str() == _("n/a"))
+      return 0;
+    return pos->toUint32(0);
   }
   pos = metadata->findKey(ExifKey("Exif.Sony2.AFAreaModeSetting"));
   if (pos != metadata->end() && pos->size() != 0 && pos->typeId() == unsignedByte) {
     std::ostringstream oss;
     pos->write(oss, metadata);
-    if (oss.str() == _("n/a")) {
-      val = 0;
-      return false;
-    }
-
-    val = pos->toUint32(0);
-    return true;
+    if (oss.str() == _("n/a"))
+      return 0;
+    return pos->toUint32(0);
   }
-
-  val = 0;
-  return false;
+  return 0;
 }
 
-static auto getMetaVersion(const ExifData* metadata, std::string& val) {
+static std::string getMetaVersion(const ExifData* metadata) {
   const auto pos = metadata->findKey(ExifKey("Exif.SonySInfo1.MetaVersion"));
-
   if (pos != metadata->end() && pos->typeId() == asciiString) {
     std::string temp = pos->toString();
-    if (temp.length() != 0) {
-      val = temp;
-      return true;
-    }
+    if (!temp.empty())
+      return temp;
   }
-  val = "";
-  return false;
+  return "";
 }
 
-static auto getFocusMode2(const ExifData* metadata, uint32_t& val) {
+static uint32_t getFocusMode2(const ExifData* metadata) {
```

Comment:
No. I just assumed there are tests for this.

Reply:
I agree with @piponazo: I'd prefer to keep the separate Boolean return value to indicate success on all of these functions, to avoid the potential confusion if the field contains `""` or `0`.

Reply:
removed.